packages/lib-ripple/index.ts

```diff
@@ -0,0 +1,68 @@
+import { DfnsApiClient, DfnsError } from '@dfns/sdk'
+import { Transaction, deriveAddress, decode, encode } from 'xrpl'
+import { GetWalletResponse, GenerateSignatureResponse } from '@dfns/sdk/types/wallets'
+
+export type DfnsWalletOptions = {
+  walletId: string
+  dfnsClient: DfnsApiClient
+}
+
+
+const assertSigned = (res: GenerateSignatureResponse) => {
+  if (res.status === 'Failed') {
+    throw new DfnsError(-1, 'signing failed', res)
+  } else if (res.status !== 'Signed') {
+    throw new DfnsError(
+      -1,
+      'cannot complete signing synchronously because this wallet action requires policy approval',
+      res
+    )
+  }
+}
+
+type WalletMetadata = GetWalletResponse
+
+export class DfnsWallet {
+  private readonly dfnsClient: DfnsApiClient
+
+  private constructor(private metadata: WalletMetadata, options: DfnsWalletOptions, private signingPubKey: string) {
+    this.dfnsClient = options.dfnsClient
+  }
+
+  public static async init(options: DfnsWalletOptions) {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Ripple' && res.network !== 'RippleTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to Ripple or RippleTestnet', { walletId, network: res.network })
+    }
+
+    // ripple adds the prefix "ed" to eddsa pubKey/private key to have
+    // the same bytes number as ecdsa pubKey.
+    const prefix = res.signingKey.scheme === 'EdDSA' ? 'ed' : ''
+
+    return new DfnsWallet(res, options, prefix + res.signingKey.publicKey)
+  }
+
+  public get address(): string {
+    return deriveAddress(this.signingPubKey)
```

Comment:
should be able to just return `this.metadata.address`.

Reply:
right, forgot that now the address is in the metadata